subprojects/core/src/main/java/org/gradle/configuration/ScriptPluginFactorySelector.java

```diff
@@ -126,4 +132,11 @@ private List<ScriptingLanguage> scriptingLanguages() {
     private ScriptPluginFactory instantiate(String provider) {
         return providerInstantiator.instantiate(provider);
     }
+
+    private void observeRemoteScriptIfPresent(ScriptSource scriptSource) {
+        URI scriptLocationUri = scriptSource.getResource().getLocation().getURI();
+        if (scriptLocationUri != null && scriptLocationUri.getScheme().startsWith("http")) {
+            remoteScriptListener.remoteScriptObserved(scriptSource);
```

Comment:
⚙️ I'd keep the callback semantics simpler, something like `scriptSourceObserved`, always call it from here and let the listener decide what to do with `scriptSource`. It shouldn't be the business of this class to make these decisions.

Reply:
Done